datamade/chi-councilmatic
# add campaign contributions to person detail

Adds a searchable table of campaign contributions to each Alder on their detail page.

This PR includes:

- adds table and section to Alder detail page
- adds `candidate_id` from IL Board of Elections for each Alder
- stores Alder metadata in a CSV and loads it in via a cached function in `utils.py`
- updates `ChicagoPerson` model code to access new metadata
- adds text and link to Illinois Sunshine page under Campaign Donations section
- pins version of `django-haystack[elasticsearch]` to v3.2.1 to handle a recently introduced build error
- removes opencivicdata 3.2 dependency. no longer needed as that [feature has been merged in](https://github.com/opencivicdata/python-opencivicdata/pull/144)

![Screenshot 2024-06-05 at 3 49 02 PM](https://github.com/datamade/chi-councilmatic/assets/919583/1b454cd2-7d71-4545-88db-dc8c9c98b9f3)

### Testing
- visit several alder pages. confirm they have a new tab to click on for Campaign Donations
- browse, search, filter the table to confirm the data looks right
- Alders without a candidate id, like the Mayor, should not show that table
